code/game/machinery/mass_driver.dm

```diff
@@ -110,6 +120,19 @@ var/list/mass_drivers = list()
 	anchored = 0
 	var/build = 0
 
+/obj/machinery/mass_driver_frame/proc/check_competition(var/turf/T = get_turf(src))
+	var/competition_found = 0
+	for(var/obj/machinery/mass_driver_frame/M in T)
```

Comment:
combine these two loops and use istype() to detect if M is either /obj/machinery/mass_driver_frame or /obj/machinery/mass_driver.